app/models/concerns/oregon_digital/generic_metadata.rb

```diff
@@ -601,52 +601,23 @@ module GenericMetadata
-      accepts_nested_attributes_for :genus, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :illustrator, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :interviewee, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :interviewer, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :local_collection_name, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :lyricist, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :military_branch, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :order, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :owner, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :patron, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :photographer, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :phylum_or_division, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :print_maker, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :publisher, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :recipient, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :repository, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :species, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :subject, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :taxon_class, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :tgn, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :transcriber, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :translator, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :cultural_context, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :style_or_period, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :form_of_work, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :institution, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :workType, reject_if: id_blank, allow_destroy: true
+
+      # Sets controlled values
+      self.controlled_properties = properties.reject { |_k, v| v.class_name.nil? }.keys.map(&:to_sym)
+
+      # Allows for the controlled properties to accept nested data
+      controlled_properties.each do |prop|
+        accepts_nested_attributes_for prop, reject_if: id_blank, allow_destroy: true
+      end
+
+      # defines a method for Generic to be able to grab a list of properties
+      define_singleton_method :controlled_props do
```

Comment:
this method should be named `controlled_properties` so that throughout the code it looks similar to how the `*_properties` methods look.

Reply:
I was afraid that would be a name collision and could effect other parts of the system. I can do more thorough testing however to see if thats actually the case.

Reply:
Stack level too deep when defining this method as #controlled_properties

Reply:
Is this because `controlled_properties` is already defined as a class_attribute in lines 603-611?

Reply:
That or because it calls `controlled_properties` in the body and it's a namespacing issue.

This might be more appropriately called `controlled_labels` anyways, since these are all `<prop>_label` values.
This might also raise the question if these should all be defined as `self.*_properties` for consistency?